exchange/src/exchange/concurrency/pin_threads.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <cstddef>
+
+#include <string>
+
+namespace nutc {
+namespace concurrency {
+
+void pin_to_core(size_t core_num, std::string thread_name = "");
```

Comment:
don't use default args

```suggestion
void pin_to_core(size_t core_num, std::string thread_name);

inline void
pin_to_core(size_t core_num)
{
    pin_to_core(core_num, "");
}
```